src/SFA.DAS.ReferenceData.PublicSectorOrgs.WebJob/Updater/IJsonManager.cs

```diff
@@ -0,0 +1,86 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Configuration;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.WindowsAzure.Storage;
+using Newtonsoft.Json;
+using SFA.DAS.NLog.Logger;
+using SFA.DAS.ReferenceData.Domain.Models;
+
+namespace SFA.DAS.ReferenceData.PublicSectorOrgs.WebJob.Updater
+{
+    public interface IJsonManager
+    {
+        void ExportFile(string filename, PublicSectorOrganisationLookUp orgs);
+
+        void UploadJsonToStorage(string filePath);
+    }
+
+    public class JsonManager : IJsonManager
+    {
+        private readonly ILog _logger;
+        private readonly string _jsonContainerName = "sfa-das-reference-data";
+        private readonly string _jsonFileName = "PublicOrganisationNames.json";
+
+        public JsonManager(ILog logger)
+        {
+            _logger = logger;
+        }
+
+        public void ExportFile(string filename, PublicSectorOrganisationLookUp orgs)
+        {
+            try
+            {
+                using (var fs = File.Open(filename, FileMode.Create))
+                {
+                    using (var sw = new StreamWriter(fs))
+                    {
+                        using (var jw = new JsonTextWriter(sw))
+                        {
+                            jw.Formatting = Formatting.Indented;
+
+                            var serializer = new JsonSerializer();
+                            serializer.Serialize(jw, orgs);
+                        }
+                    }
+                }
+            }
+            catch (Exception ex)
+            {
+                _logger.Error(ex, $"An error occurred exporting data to {filename}: {ex.Message}");
+                throw new Exception($"An error occurred exporting data to {filename}: {ex.Message}");
```

Comment:
We're losing the exception type here if we throw an exception. It may come in as an ArgumentException and go out as Exception.